### DISCUSSION TITLE
Text printed with "Back" menu item

### DISCUSSION THREAD

**Discussion by jeroen94704** [davetcc/tcMenu#430] [Q&A] [answered]
I just found out about tcmenu and so far I'm very happy with it! 

One question I have is about the "Back" menu item that is silently added to submenu's. The documentation states this item will display as "[Back..]". However, in my menu it is displayed as "\<NAME OF THE SUBMENU\> [..]". Looking at the code this seems to be controlled by the RenderFnMode parameter passed into "backSubItemRenderFn" in RuntimeMenuItem.cpp. If this has the value "RENDERFN_NAME" it will construct the string as I see it. I would like the behavior as implemented by RENDERFN_VALUE, which would simply display "[..]". It is easy to change this in code, of course, but this all seems very deliberate, so I what is the proper way to change this?

Thanks in advance!

Jeroen

**Comment by davetcc** [comment, marked as answer]
I mean I've never really considered this much, you can override the rendering of the back menu, but effectively the back menu from a theme perspective is just a title menu. Overriding how particular elements are drawn is documented here https://www.thecoderscorner.com/products/arduino-libraries/tc-menu/themes/rendering-with-themes-icons-grids/

**Reply by jeroen94704** [reply]
Hm, I'll have to figure out how themes work then. I just thought it was odd that the "back" functionality is combined with the title bar. From a UX perspective it's not necessarily obvious that a title bar functions like yet another menu item, so I figured I must be doing something wrong. 

But thanks for your answer, it definitely helps!

**Reply by jeroen94704** [reply]
By the way, I'd like to sponsor this project, but do you have any means other than GitHub for that (e.g. Patreon, KoFi)? I don't feel comfortable handing over my personal details (they require address and everything) just for this.

**Reply by davetcc** [reply]
Once the current release is out, the next thing I'm going to do is tidy up themes, making it far easier to define and work with them, using some kind of builder syntax.

In terms of other sponsor systems, agreed, I need to look at maybe supporting a couple more options.
